synapse/storage/schema/main/delta/78/02_validate_and_update_user_filters.py

```diff
@@ -74,6 +72,12 @@ def run_upgrade(
         """
         cur.execute(index_sql)
 
+        full_user_id_idx_sql = """
+        CREATE UNIQUE INDEX IF NOT EXISTS user_filters_full_user_id_unique ON
+            temp_user_filters (full_user_id, filter_id)
+        """
```

Comment:
Hmm, I think we need to add this to a new delta, otherwise for deployments where this file successfully run previously they'll end up with a different schema?

Reply:
And we'll also potentially have to drop the incorrect indexes.

Reply:
> And we'll also potentially have to drop the incorrect indexes.

This is referring to the indexes that were automatically created on the columns when the `UNIQUE` constraint was declared, correct?

Reply:
Errr I just mean we'll need to undo the results of the bad upgrade, yes.